src/OpenLoco/Objects/ObjectManager.cpp

```diff
@@ -998,13 +1002,88 @@ namespace OpenLoco::ObjectManager
+        const auto installedObjects = getAvailableObjects(header.getType());
+        auto res = std::find_if(std::begin(installedObjects), std::end(installedObjects), [&header](auto& obj) { return *obj.second._header == header; });
+        if (res == std::end(installedObjects))
+        {
+            // Object is not installed
+            return false;
+        }
+
+        const auto& installedObject = res->second;
+        const auto filePath = Environment::getPath(Environment::PathId::objects) / fs::u8path(installedObject._filename);
+
+        SawyerStreamReader stream(filePath);
+        ObjectHeader loadingHeader;
+        stream.read(&loadingHeader, sizeof(loadingHeader));
+        if (loadingHeader != header)
+        {
+            // Something wrong has happened and installed object does not match index
+            // Vanilla continued to search for subsequent matching installed headers.
+            return false;
+        }
+
+        // Vanilla would branch and perform more efficient readChunk if size was known from installedObject.ObjectHeader2
+        auto data = stream.readChunk();
+
+        if (!computeObjectChecksum(loadingHeader, data))
+        {
+            // Something wrong has happened and installed object checksum is broken
+            return false;
+        }
+
+        // Copy the object into Loco freeable memory (required for when load loads the object)
+        auto* object = reinterpret_cast<Object*>(malloc(data.size()));
+        std::copy(std::begin(data), std::end(data), reinterpret_cast<uint8_t*>(object));
+
+        if (!callObjectFunction(loadingHeader.getType(), *object, ObjectProcedure::validate))
+        {
+            free(object);
+            object = nullptr;
+            // Object failed validation
+            return false;
+        }
+
+        if (_totalNumImages >= Gfx::G1ExpectedCount::kObjects + Gfx::G1ExpectedCount::kDisc)
+        {
+            free(object);
+            object = nullptr;
+            // Too many objects loaded and no free image space
+            return false;
+        }
+
+        // sub_471BCE only code
+        //for (id = 0; id < getMaxObjects(loadingHeader.getType()); ++id)
+        //{
+        //    if (object_repository[enumValue(loadingHeader.getType())].objects[id] == reinterpret_cast<Object*>(-1))
+        //    {
+        //        break;
+        //    }
+        //}
+        //if (id >= getMaxObjects(loadingHeader.getType()))
+        //{
+        //    free(object);
+        //    return false;
+        //}
+
+        object_repository[enumValue(loadingHeader.getType())].objects[id] = object;
+        auto& extendedHeader = object_repository[enumValue(loadingHeader.getType())].object_entry_extendeds[id];
+        extendedHeader = ObjectEntry2{
+            loadingHeader, data.size()
+        };
+
+        if (!*_isPartialLoaded)
+        {
+            callLoadObjectFunction({ loadingHeader.getType(), id }, *object);
```

Comment:
it looks like `callObjectFunction` can be passed an `ObjectProcedure`, but for `ObjectProcedure::load` it looks like we should really call `callLoadObjectFunction` instead, so should we make it assert/fail if you call `callObjectFunction` with the procedure as `load`?

Reply:
These are all basically stop gaps until we implement these functions. I have started work on implementing them and removing the need for these callers. Yeah could add an assert